webcrypto/subtle_crypto.go

```diff
@@ -234,8 +234,37 @@ func (sc *SubtleCrypto) Digest(algorithm goja.Value, data goja.Value) *goja.Prom
 //
 // The `keyUsages` parameter is an array of strings indicating what the key can be used for.
 func (sc *SubtleCrypto) GenerateKey(algorithm goja.Value, extractable bool, keyUsages []CryptoKeyUsage) *goja.Promise {
-	// TODO: implementation
-	return nil
+	promise, resolve, reject := sc.makeHandledPromise()
+
+	normalized, err := normalizeAlgorithm(sc.vu.Runtime(), algorithm)
+	if err != nil {
+		reject(err)
+		return promise
+	}
+
+	go func() {
+		switch normalized.Name {
+		case AESCbc, AESCtr, AESGcm, AESKw:
+			params := &AesKeyGenParams{
+				Algorithm: normalized,
+				Length:    algorithm.ToObject(sc.vu.Runtime()).Get("length").ToInteger(),
```

Comment:
This is accessing the goja runtime not on the event loop. While this particualr case *might* not blow up in this particular case it is arguably better to just get the `length` before the goroutine or have a differnt way to acquire it that doesn't not access goja objects off the event loop